cloud/deployment/deployment.go

```diff
@@ -719,28 +770,319 @@ func Update(deploymentID, label, ws, description, deploymentName, dagDeploy, exe
 				"\nRun `astro deploy` after this command to restart your DAGs. It may take a few minutes for the Airflow UI to update.")
 			confirmWithUser = true
 		}
-		deploymentUpdate.DagDeployEnabled = false
+		dagDeployEnabled = false
 	}
 
-	// if we have worker queues add them to the input
-	if len(wQueueList) > 0 {
+	configOption, err := GetDeploymentOptions("", astrocore.GetDeploymentOptionsParams{}, coreClient)
+	if err != nil {
+		return err
+	}
+
+	// build query input
+	updateDeploymentRequest := astroplatformcore.UpdateDeploymentRequest{}
+
+	if name == "" {
+		name = currentDeployment.Name
+	}
+	if description == "" {
+		if currentDeployment.Description != nil {
+			description = *currentDeployment.Description
+		}
+	}
+	var deploymentEnvironmentVariablesRequest []astroplatformcore.DeploymentEnvironmentVariableRequest
+	if len(newEnvironmentVariables) == 0 {
+		if currentDeployment.EnvironmentVariables != nil {
+			EnvironmentVariables := *currentDeployment.EnvironmentVariables
+			for i := range *currentDeployment.EnvironmentVariables {
+				var deploymentEnvironmentVariableRequest astroplatformcore.DeploymentEnvironmentVariableRequest
+				deploymentEnvironmentVariableRequest.IsSecret = EnvironmentVariables[i].IsSecret
+				deploymentEnvironmentVariableRequest.Key = EnvironmentVariables[i].Key
+				deploymentEnvironmentVariableRequest.Value = EnvironmentVariables[i].Value
```

Comment:
best practice to keep these variable names camelCase 
```suggestion
			environmentVariables := *currentDeployment.EnvironmentVariables
			for i := range environmentVariables {
				var deploymentEnvironmentVariableRequest astroplatformcore.DeploymentEnvironmentVariableRequest
				deploymentEnvironmentVariableRequest.IsSecret = environmentVariables[i].IsSecret
				deploymentEnvironmentVariableRequest.Key = environmentVariables[i].Key
				deploymentEnvironmentVariableRequest.Value = environmentVariables[i].Value
```

Reply:
fixed